src/Aspire.ProjectTemplates/templates/aspire-starter/.template.config/template.json

```
@@ -84,12 +84,26 @@
       "displayName": "_Use Redis for caching (requires a supported container runtime)",
       "description": "Configures whether to setup the application to use Redis for caching. Requires a supported continer runtime to run locally, see https://aka.ms/dotnet/aspire/containers for more details."
     },
-    "CreateTestsProject": {
+    "TestProjectFrameworkKind": {
       "type": "parameter",
-      "datatype": "bool",
-      "defaultValue": "false",
+      "datatype": "choice",
+      "defaultValue": "None",
       "displayName": "Create a _tests project",
-      "description": "Configures whether to create a project for integration tests using the AppHost project."
+      "description": "Configures whether to create a project for integration tests using the AppHost project.",
+      "choices": [
+        {
+          "choice": "None",
+          "description": "Do not create a test project."
+        },
+        {
+          "choice": "MSTest",
+          "description": "Creates a MSTest test project using the AppHost project."
+        },
+        {
+          "choice": "xUnit",
+          "description": "Creates a xUnit test project using the AppHost project."
```

I think the official name is "xUnit.net" so suggest we go with:
```suggestion
          "choice": "xUnit.net",
          "description": "Creates an xUnit.net test project using the AppHost project."
```

---

> You are obviously better at English than me, I'm just wondering isn't the rule to have 'a' before consonant and 'an' before voyelle?

I believe that's generally correct, but 'x' is pronounced as 'ex', so that would result in 'an'. I'm not native English however, so I might be mistaken about this exception.

---

@thimok is correct. The rule is actually based on the sound of the following word, not the actual letters used: https://www.merriam-webster.com/grammar/is-it-a-or-an